core/src/main/java/org/apache/calcite/adapter/jdbc/JdbcRules.java

```diff
@@ -105,8 +105,8 @@ private JdbcRules() {
       };
 
   static final RelFactories.FilterFactory FILTER_FACTORY =
-      (input, condition) -> new JdbcRules.JdbcFilter(input.getCluster(),
-          input.getTraitSet(), input, condition);
+      (input, condition, correlVariables) -> new JdbcRules.JdbcFilter(
```

Comment:
Shouldn't we check if correlVariables is empty? (same as EnumerableFilter...)

Reply:
Done